experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -530,22 +573,54 @@ export class EditManager<
 			this.trunkUndoRedoManager.trackCommit(trunkHead, type);
 		}
 		this.trunk.repairDataStoreProvider?.applyChange(commit.change);
-		this.sequenceMap.set(sequenceNumber, trunkHead);
-		this.trunkMetadata.set(trunkHead.revision, { sequenceNumber, sessionId: commit.sessionId });
+		this.sequenceMap.set(sequenceId, trunkHead);
+		this.trunkMetadata.set(trunkHead.revision, { sequenceId, sessionId: commit.sessionId });
 		this.events.emit("newTrunkHead", trunkHead);
 	}
 
 	/**
-	 * Finds the most recent trunk commit that was sequenced at or before the given sequence number.
+	 * Finds the most recent trunk commit that was sequenced at or before the given sequence number. If the commit was part of a batch an index can be pass.
 	 * @param sequenceNumber - the sequence number to search for
+	 * @param indexInBatch - the commit index inside the batch to search for in conjunction with the sequence number
 	 * @remarks Fails if there is no eligible commit.
-	 * @returns the closest commit and its sequence number
+	 * @returns the closest commit and its sequence id
 	 */
-	private getClosestTrunkCommit(sequenceNumber: SeqNumber): [SeqNumber, GraphCommit<TChangeset>] {
-		const commit = this.sequenceMap.getPairOrNextLower(sequenceNumber);
-		assert(commit !== undefined, 0x72a /* Sequence number has been evicted */);
+	private getClosestTrunkCommit(
```

Comment:
You could try splitting this into two overloads:
`getClosestTrunkCommit(sequenceNumber: SeqNumber)`
`getClosestTrunkCommit(sequenceId: SequenceId)`

It might make it a little easier to document and/or implement. It also might not (*shrug*).

Reply:
In order to this approach work shouldn't SeqNumber and SequenceId be compatible types, or I have to add a 3rd signature witha param with SeqNumber | SequenceId type?

Reply:
Right - [function overloads in typescript](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) consist of the signature declarations followed by the implementation declaration (which is not callable). So the callable signatures would be what I suggested, but the implementation signature would take a `SeqNumber | SequenceId`. In the function body, if you check `typeof parameter === number`, typescript will infer `SeqNumber` in the `true` case and `SequenceId` in the false case. There isn't a super obvious need for an overload here, but it might just be a fun exercise if you haven't written overloads much. They can be useful in lots of ways, for example [this](https://www.typescriptlang.org/play?#code/PTAEDEFcDsGMBcCWB7apkDcCmAnANsgIYAmi0A5qLIWmQGZ6RZxahnxZ56IDOzfAOgBQIUADlkHUAAtkAd1DxprAAZ0VoGsVAryGujAQpoPUACNO80KTp1cbNEqyIcDjl179WOLD0h54HgAaUCxsR1lIcmlFZRcqQi5CMzxWHkRyaEJ4SB9TQh82YmYkajxhIQM4JFRQOgAKAEoALlAMZERiAG5Kwxq0BuhW6EgAWwscFraO7t7q4zr66AB+YbGJqfbO0ABvIVAD0FEBE6EAXyE5o1ryJdXQEfHcTZnd-cPj04uRMAbD0AAPABaIGgADKUXIvkCijkyFA6Uy2VyvhCqFYckQSk0oAADgVCKMsBxXOj0NAMVjIvABKAACrKBFYWCoYghUbIQos0a47jUDjaZC4-psUzSTrFaDCUS3f7A0EQ8hQngwsmIrI5HwhOTKQpOdDC4yJPEEokk0VtXiIFKsOg4ZCjWJpeAFGlCIA) or [this](https://www.typescriptlang.org/play?#code/PTAEDEFcDsGMBcCWB7apkDcCmAnANsgIYAmi0A5qLIWtITjsgO6g5byQ5rwCeADlgDOAOgBQIUAGFUgxMVyhCoAGYwEKNEwAWiWFqpaa5IVULxBoMvGShiychZrFb9i1ZvVzALlGjVcJFQDIywACk8vUAAiTyiASkiou3IogG4-NUC0PRDQ5MTk+MTY9P91IJyKMJpEAFtCPGKzKNAAH2jChI77FvaY5tAAb1FQUctlUFCa+rxQAF4F6Ni4oZGx9bYOLm6U9PWAX1AsPEEsVfWN9k40fvg0tdH90SfxMAAVLTOyrMtavjwsLUsNB4GYNIpYLAsHxzKAAAa3Xo7KJwxTQZyba4WBGxNrIuEAGlA2l0+iYyEgeGc0GQOBmeB4rCwAEdIIg2KB4J9THgAThOR5CIJ4JzuWxBJT4GIJB8zphcAQSI5ecxRWdMdteAILNZQAAjM50BjMLDOPVC03objcsh8SAirVCMSwGQizzzYJVUJJHpxVKgCSO5ATd0LOZLZqiF3QYUuSjhyrGb3Lf2B-hYYNx+aLH0pIA). :)